ui/apps/everest/src/pages/settings/namespaces/namespace-details/namespace-details.tsx

```diff
@@ -0,0 +1,206 @@
+} from 'shared-types/dbEngines.types';
+import { useNamespace } from 'hooks/api/namespaces';
+import { useDbEngines, useOperatorUpgrade } from 'hooks/api/db-engines';
+import { NoMatch } from 'pages/404/NoMatch';
+import { FormProvider, useForm } from 'react-hook-form';
+import UpgradeHeader from './upgrade-header';
+import ClusterStatusTable from './cluster-status-table';
+import UpgradeModal from './upgrade-modal';
+import { getOperatorUpgradePreflight } from 'api/dbEngineApi';
+import EngineLowerContent from './engine-lower-content';
+
+const NamespaceDetails = () => {
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const [modalOpen, setModalOpen] = useState(false);
+  const { namespace: namespaceName = '' } = useParams();
+  const [selectedEngineIdx, setSelectedEngineIdx] = useState<number>();
+  const { data: namespace, isLoading: loadingNamespace } = useNamespace(
+    namespaceName,
+    {
+      enabled: !!namespaceName,
+    }
+  );
+  const { data: dbEngines = [] } = useDbEngines(
+    namespaceName,
+    {
+      enabled: !!namespace,
+      refetchInterval: 2 * 1000,
+    },
+    true
+  );
+
+  const methods = useForm({
+    defaultValues: {
+      dbType: dbEngineToDbType(dbEngines[0]?.type),
+    },
+  });
+
+  const selectedEngine: DbEngine | null =
+    selectedEngineIdx !== undefined ? dbEngines[selectedEngineIdx] : null;
+
+  const targetVersion =
+    !!selectedEngine && selectedEngine.pendingOperatorUpgrades?.length
+      ? selectedEngine?.pendingOperatorUpgrades[0].targetVersion
+      : '';
+
+  const someEngineIsUpgrading = dbEngines.some(
+    (engine) => engine.status === DbEngineStatus.UPGRADING
+  );
+
+  const preflightQueriesResults = useQueries({
+    queries: dbEngines.map((engine) => ({
+      queryKey: ['dbEngineUpgradePreflight', namespaceName, engine.name],
+      refetchInterval: 2 * 1000,
+      queryFn: () =>
+        engine.pendingOperatorUpgrades?.length
+          ? getOperatorUpgradePreflight(
+              namespaceName,
+              engine.name,
+              engine.pendingOperatorUpgrades[0].targetVersion
+            )
+          : Promise.resolve<OperatorUpgradePreflightPayload>({
+              currentVersion: engine.operatorVersion,
+              databases: undefined,
+            }),
+      enabled: !!namespace && !!dbEngines.length && !someEngineIsUpgrading,
+    })),
+  });
+
+  const { mutate: upgradeOperator } = useOperatorUpgrade(
+    namespaceName,
+    selectedEngine?.name || '',
+    targetVersion
+  );
+  const lastTargetVersion = selectedEngine?.pendingOperatorUpgrades?.length
+    ? selectedEngine?.pendingOperatorUpgrades[
+        selectedEngine?.pendingOperatorUpgrades?.length - 1
+      ].targetVersion
+    : '';
+
+  const performUpgrade = (engineName: string) => {
+    upgradeOperator(null, {
+      onSuccess: () => {
+        queryClient.setQueryData<GetDbEnginesPayload>(
+          [`dbEngines_${namespace}`],
```

Comment:
could we use a simple list here, instead of a template? like we do for [dbs/backups](https://github.com/percona/everest/blob/main/ui/apps/everest/src/hooks/api/db-cluster/useDbActions.ts#L33)